Newsletters: add 281 (2023-12-13) (#1428)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2023-12-13-newsletter.md

---
title: 'Bitcoin Optech Newsletter #281'
permalink: /en/newsletters/2023/12/13/
name: 2023-12-13-newsletter
slug: 2023-12-13-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

> - **Discussion about griefing liquidity ads:** Bastien Teinturier

Reviewer: Totally optional, but readers could be interested in some our chat with Lisa/tbast a couple weeks ago? (could just be a link). https://bitcoinops.org/en/podcast/2023/11/30/#update-to-the-liquidity-ads-specification-transcript

  [posted][teinturier liqad] to the Lightning-Dev mailing list about a

>   potential problem with timelocks on [dual funded channels][topic dual

Reviewer 2: ```suggestion
  potential problem with timelocks on [dual-funded channels][topic dual
```

  funding] created from [liquidity advertisements][topic liquidity
  advertisements].  For example, Alice advertises that, for a fee, she's
  willing to commit 10,000 sats of her funds to a channel for 28 days.

>   The 28 day timelock prevents Alice from simply closing the channel

Reviewer 2: ```suggestion
  The 28-day timelock prevents Alice from simply closing the channel
```

  after she receives payment and using her funds for something else.

    Continuing the example, Bob opens the channel with an additional
    contribution of 100,000,000 sats (1 BTC) of his funds.  He then
    sends almost all of his funds through the channel.  Now Alice's
    balance in the channel isn't the 10,000 sats she received a fee
    for---it's almost 10,000 times higher than that amount.  If Bob is
    malicious, he won't allow those funds to move again until the

>     expiration of the 28 day timelock to which Alice committed.

Reviewer 2: ```suggestion
    expiration of the 28-day timelock to which Alice committed.
```

    A mitigation suggested by Teinturier and discussed by him and others

>     was attempting to only encumber the liquidity contribution by the

Reviewer 2: ```suggestion
    was attempting to encumber the liquidity contribution using only the
```

    timelock (e.g., only Alice's 10,000 sats).  This introduces

>     complexitise and inefficiencies, although it may solve the problem.

Reviewer 3: ```suggestion
    complexities and inefficiencies, although it may solve the problem.
```

>     complexitise and inefficiencies, although it may solve the problem.

Reviewer 2: ```suggestion
    complexity and inefficiency, although it may solve the problem.
```

- **BBQr encoding scheme announced:**
  The [scheme][bbqr github] can efficiently encode larger files, for example [PSBTs][topic
  psbt], into an animated QR series for use in air-gapped wallet configurations.

- **Zeus v0.8.0 released:**
  The [v0.8.0][zeus v0.8.0] release contains an embedded LND node, additional
  [zero conf channel][topic zero-conf channels] support, and support for simple taproot channels,
  among other changes.

## Selected Q&A from Bitcoin Stack Exchange

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aa
nswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [What are all the rules related to CPFP fee bumping?]({{bse}}120853)
  Pieter Wuille points out that contrary to the [RBF][topic rbf] fee bumping
  technique that has a list of associated policy rules, the [CPFP][topic cpfp]

>   fee bumping technique has no additional policy rules.

Reviewer 2: This isn't a comment on the PR here, but mention of "policy" reminds me that recently I saw a useful distinction that I wasn't aware of, it may have been in a Murch tweet (sorry, post on X). Transaction "standardness" relates to properties of a single transaction considered in isolation; "policy" involves multiple transactions (ancestor and descendant counts, for example). Previously, I had thought of these two words as somewhat interchangeable. Just thought I'd pass that along in case others here weren't aware of those definitions!

Author: I don't think that's quite right.  I agree that standardness relates to the properties of a single transaction (inclusive of its prevouts),  but I consider policy to be a superset of standardness rules that also includes all other rules related to whether a node will accept a transaction into its mempool and relay it.  I think this is evidenced by `IsStandard()` and `IsStandardTx` being defined in `policy.*`.

  context of [BIP125][]’s rule 5: “The number of original transactions to be
  replaced and their descendant transactions which will be evicted from the
  mempool must not exceed a total of 100 transactions”. Readers may also be
  interested in the [Add BIP-125 rule 5 testcase with default
  mempool][review club 25228] PR Review Club meeting.

- [What types of RBF exist and which one does Bitcoin Core support and use by default?]({{bse}}120749)
  Murch provides some of Bitcoin Core’s transaction replacement history and in a
  [related question]({{bse}}120773), a summary of RBF replacement rules and
  links to Bitcoin Core’s [Mempool Replacements][bitcoin core mempool
  replacements] documentation and one developer's ideas for [RBF
  improvements][glozow rbf improvements].

- [What is the Block 1,983,702 Problem?]({{bse}}120834)
  Antoine Poinsot gives an overview of the issues that led to [BIP30][]
  restricting duplicate txids and [BIP34][] mandating the inclusion of the
  current block height in the coinbase field. He then points out that there are
  numerous blocks whose random coinbase field content happens to match the

>   mandatory height prefix of a later block. Block 1 983 702 being the first

Reviewer 2: ```suggestion
  mandatory height prefix of a later block. Block 1 983 702 is the first
```
Also, I don't think I've see spaces within number groups like this, maybe should be commas?

  for which it would be practically possible to repeat the coinbase transaction

>   of the prior block. In a [related question]({{bse}}120836) Murch and

Reviewer 2: ```suggestion
  of the prior block. In a [related question]({{bse}}120836), Murch and
```

  Antoine Poinsot evaluate that possibility in greater detail.

- [What are hash functions used for in bitcoin?]({{bse}}120418)
  Pieter Wuille lists over thirty different instances across consensus rules,

>   peer to peer protocol, wallet and node implementation details that make use

Reviewer 2: ```suggestion
  peer-to-peer protocol, wallet and node implementation details that make use
```

  of no less than 10 different hash functions.

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [LND 0.17.3-beta.rc1][] is a release candidate that contains several
  bug fixes.

<!-- FIXME:harding to update Tuesday -->

## Notable code and documentation changes

*Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo], and
[Bitcoin Inquisition][bitcoin inquisition repo].*

- [LDK #2685][] adds the ability to obtain block chain data from an
  Electrum-style server.

- [Libsecp256k1 #1446][] removes some x86_64 assembly code from the
  project, switching to using existing C language code that has always

>   been used for other platforms.  The assembly code was human optimized

Reviewer 2: ```suggestion
  been used for other platforms.  The assembly code was human-optimized
```

>   several years to improve performance but, in the meantime, compilers

Reviewer 2: ```suggestion
  over several years to improve performance but, in the meantime, compilers
```
(Or "for several years" or maybe even "several years ago"?)

  improved and recent versions of both GCC and LLVM (clang) now produce
  even more performant code.

- [BTCPay Server #5389][] adds support for [BIP129][] secure multisig

>   wallet setup (see [Newsletter #136][news136 bip129].  This allows

Reviewer 3: ```suggestion
  wallet setup (see [Newsletter #136][news136 bip129]).  This allows
```

  BTCPay server to interact with multiple software wallets and hardware
  signing devices as part of a simple coordinated multisig setup
  procedure.

- [BTCPay Server #5490][] begins using [fee estimates][ms fee api] from

>   [Mempool.space][] by default, with a fallback on fee estimates from

Reviewer: ```suggestion
  [mempool.space][] by default, with a fallback on fee estimates from
```
